pkg/agent/cache/credentialrequestdefinitions.go

```diff
@@ -0,0 +1,33 @@
+package cache
+
+import v1 "github.com/Axway/agent-sdk/pkg/apic/apiserver/models/api/v1"
+
+// Credential request definition cache management
+
+// AddCredentialRequestDefinition -  add/update CredentialRequestDefinition resource in cache
+func (c *cacheManager) AddCredentialRequestDefinition(resource *v1.ResourceInstance) {
+	defer c.setCacheUpdated(true)
+
+	c.crdMap.Set(resource.Name, resource)
```

Comment:
To make caches consistent can we rather use Metadata.ID as the key and use name as secondary key?. Also add methods Get method(default to return by ID)?